pkg/distributor/http.go

```diff
@@ -72,11 +73,16 @@ func ParseRequest(r *http.Request) (*logproto.PushRequest, error) {
 			totalEntries     int64
 		)
 
+		mostRecentEntry := time.Unix(0, 0)
+
 		for _, s := range req.Streams {
 			streamLabelsSize += int64(len(s.Labels))
 			for _, e := range s.Entries {
 				totalEntries++
 				entriesSize += int64(len(e.Line))
+				if e.Timestamp.After(mostRecentEntry) {
```

Comment:
should we just look at the last entry for each stream instead ?

Reply:
```go
if len(s.Entries) > 0{
  if s.Entries[len(s.Entries)-1].Timestamp.After(mostRecentEntry) {
    mostRecentEntry = s.Entries[len(s.Entries)-1).Timestamp
}
```

Reply:
if/when we support out of order the current implementation would handle that a little better but I agree this is a better optimization for our current constraints